qa/L0_backend_python/io/io_test.py

```diff
@@ -162,6 +162,80 @@ def test_variable_gpu_output(self):
                 self.assertEqual(output.size, i + 1)
                 np.testing.assert_almost_equal(output, np.ones(i + 1) * (i + 1))
 
+    # Non-decoupled models should filter outputs base on requested outputs.
+    def test_requested_output_default(self):
+        model_name = "add_sub"
+        shape = [16]
+
+        input0_data = np.random.rand(*shape).astype(np.float32)
+        input1_data = np.random.rand(*shape).astype(np.float32)
+        inputs = [
+            grpcclient.InferInput(
+                "INPUT0", input0_data.shape, np_to_triton_dtype(input0_data.dtype)
+            ),
+            grpcclient.InferInput(
+                "INPUT1", input1_data.shape, np_to_triton_dtype(input1_data.dtype)
+            ),
+        ]
+        inputs[0].set_data_from_numpy(input0_data)
+        inputs[1].set_data_from_numpy(input1_data)
+
+        # request for output 1, among output 0 and 1.
+        requested_outputs = [grpcclient.InferRequestedOutput("OUTPUT1")]
+
+        with self._shm_leak_detector.Probe():
+            response = self._client.infer(
+                model_name=model_name,
+                inputs=inputs,
+                outputs=requested_outputs,
+            )
+
+        outputs = response.get_response().outputs
+        self.assertEqual(len(outputs), len(requested_outputs))
+        output1_data = response.as_numpy("OUTPUT1")
+        self.assertTrue(np.allclose(input0_data - input1_data, output1_data))
+
+    # Decoupled models currently do not filter outputs base on requested outputs.
+    def test_requested_output_decoupled(self):
+        model_name = "dlpack_io_identity_decoupled"
+        shape = [4]
+
+        input0_data = np.random.rand(*shape).astype(np.float32)
+        gpu_output_data = np.random.rand(*shape).astype(np.bool_)
+        inputs = [
+            grpcclient.InferInput(
+                "INPUT0", input0_data.shape, np_to_triton_dtype(input0_data.dtype)
+            ),
+            grpcclient.InferInput(
+                "GPU_OUTPUT",
+                gpu_output_data.shape,
+                np_to_triton_dtype(gpu_output_data.dtype),
+            ),
+        ]
+        inputs[0].set_data_from_numpy(input0_data)
+        inputs[1].set_data_from_numpy(gpu_output_data)
+
+        # request for output 0, among output 0 and next gpu output.
+        requested_outputs = [grpcclient.InferRequestedOutput("OUTPUT0")]
```

Comment:
If we already have the test and just removing that condition would fix decoupled mode as well, let's update the test to make it work for decoupled mode as well.

Reply:
Sure, [Reverse decoupled requested output filtering test goal](https://github.com/triton-inference-server/server/pull/7338/commits/1388172906cd2844588402fa20a282b4959591ff)

Reply:
We probably have tested that in other test already, do we have test to make sure "not specifying requested output returns all outputs"?

Reply:
We do have it in other tests, but not explicitly test for it. It is an easy addition into the existing two tests, for a complete testing of the requested output behavior: [Add verification for all outputs returned without requested output](https://github.com/triton-inference-server/server/pull/7338/commits/b6e10f82b29840fe5700824fc8072db1edc123f1)